### DISCUSSION TITLE
Any documentation on hosting locally on a LXC or bare metal server?

### DISCUSSION THREAD

**Discussion by ithakaa** [notea-org/notea#65] [Q&A]
Looking for a note taking app I can self host, do we have documentation for shelf hosting?

**Comment by QingWei-Li** [comment]
It is easy to install via docker, as mentioned in the documentation. https://github.com/QingWei-Li/notea#docker

**Comment by ithakaa** [comment]
> It is easy to install via docker, as mentioned in the documentation. https://github.com/QingWei-Li/notea#docker

I'm not looking for host it in a docker
Any chance we can get documentation ?

**Comment by QingWei-Li** [comment]
Do you mean to run the app service directly on your server?

Maybe you need to install NodeJS and clone the repo on your machine. 
Next create `.env` file in the repo directory https://github.com/QingWei-Li/notea#storage
Compile code via `yarn && yarn build`
Run service via `yarn start`

**Comment by ithakaa** [comment]
> Do you mean to run the app service directly on your server?
> 
> Maybe you need to install NodeJS and clone the repo on your machine.
> Next create `.env` file in the repo directory https://github.com/QingWei-Li/notea#storage
> Compile code via `yarn && yarn build`
> Run service via `yarn start`

I'm so so sorry to keep pestering you. 
I'm not a developer or experienced Linux guru
If you can help with a small readme, it would be really good

**Comment by QingWei-Li** [comment]
Don't you consider deploying to vercel or netilfy? This is the easiest and most efficient way.


Click the button [![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https%3A%2F%2Fgithub.com%2FQingWei-Li%2Fnotea&env=STORE_ACCESS_KEY,STORE_SECRET_KEY,STORE_BUCKET,STORE_END_POINT,STORE_FORCE_PATH_STYLE,PASSWORD)

Then set the environment variable refer to this [example](https://github.com/QingWei-Li/notea#docker) The premise is that you have an s3 storage service and know how to find these values.


![image](https://user-images.githubusercontent.com/7565692/119126980-4f039f00-ba66-11eb-914a-49525bed8789.png)

**Comment by ithakaa** [comment]
> Don't you consider deploying to vercel or netilfy? This is the easiest and most efficient way.
> 
> Click the button [![Deploy with Vercel](https://camo.githubusercontent.com/5e471e99e8e022cf454693e38ec843036ec6301e27ee1e1fa10325b1cb720584/68747470733a2f2f76657263656c2e636f6d2f627574746f6e)](https://vercel.com/new/git/external?repository-url=https%3A%2F%2Fgithub.com%2FQingWei-Li%2Fnotea&env=STORE_ACCESS_KEY,STORE_SECRET_KEY,STORE_BUCKET,STORE_END_POINT,STORE_FORCE_PATH_STYLE,PASSWORD)
> 
> Then set the environment variable refer to this [example](https://github.com/QingWei-Li/notea#docker) The premise is that you have an s3 storage service and know how to find these values.
> 
> ![image](https://user-images.githubusercontent.com/7565692/119126980-4f039f00-ba66-11eb-914a-49525bed8789.png)

I would honestly like to host it on my local home network.
Sorry for the trouble

**Reply by QingWei-Li** [reply]
I found a way use [nexe](https://github.com/nexe/nexe) to package the node application into binary, and I did it.

I want to know what operating system you use? I can pack a copy for you to try. If it's work, I plan to add this feature.

![image](https://user-images.githubusercontent.com/7565692/119210533-36839b00-badf-11eb-9856-948d3fd54030.png)

**Comment by ithakaa** [comment]
I use Ubuntu Server 20
Thanks for looking I to this for me

**Reply by QingWei-Li** [reply]
The currently packaged executable has bugs, but it works. It contains a `.env` file, you need to configure your own s3 service.

Open the executable file and visit `localhost:3000`

https://drive.google.com/file/d/1frvqG-Ocz8QAo48EDI1gd8mei16XEYg5/view?usp=sharing
